### PR TITLE
ironware_facts module

### PR DESCRIPTION
**Pull request by paulquack**
##### SUMMARY

Facts module for Brocade Ironware routers

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

ironware_facts

##### ANSIBLE VERSION

```
ansible 2.5.0 (ironware_facts 83955157c0) last updated 2017/10/26 22:18:00 (GMT +1100)
  config file = /Users/paul.baker/.ansible.cfg
  configured module search path = [u'/Users/paul.baker/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/paul.baker/ansible/lib/ansible
  executable location = /Users/paul.baker/ansible/bin/ansible
  python version = 2.7.13 (default, Dec 18 2016, 07:03:39) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/network/ironware/ironware_facts.py [hunk 1]
```diff
@@ -0,0 +1,684 @@
...
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (ganeshrn) [inline lib/ansible/modules/network/ironware/ironware_facts.py]: Please use update bolierplate.

File: lib/ansible/modules/network/ironware/ironware_facts.py [hunk 2]
```diff
@@ -0,0 +1,684 @@
...
+#
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
```
Comment (ganeshrn) [inline lib/ansible/modules/network/ironware/ironware_facts.py]: This should be `'supported_by': 'community'`

File: lib/ansible/modules/network/ironware/ironware_facts.py [hunk 3]
```diff
@@ -0,0 +1,684 @@
...
+EXAMPLES = """
+# Note: examples below use the following provider dict to handle
+#       transport and authentication to the node.
+---
+vars:
+  cli:
+    host: "{{ inventory_hostname }}"
+    username: username
+    password: secret
+    transport: cli
```
Comment (ganeshrn) [inline lib/ansible/modules/network/ironware/ironware_facts.py]: Same as the comment for `ironware_config` module. As `transport` is not part of provider spec, setting it in provider will result in error. The long-term goal is to move away from the provider.

Comment (paulquack) [reply lib/ansible/modules/network/ironware/ironware_facts.py]: `provider` removed as per `ironware_config` module.

**Comment by paulquack** [comment]
Thanks @ganeshrn

**Comment by paulquack** [comment]
shipit

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Thank you for the module and tests. 
Merged into `devel`, will be released in  Ansible 2.5.0.

**Pull request merged**